src/bleak_esphome/backend/client.py

```diff
@@ -264,20 +276,30 @@ def _on_bluetooth_connection_state(
 
     @api_error_as_bleak_error
     async def connect(
-        self, dangerous_use_bleak_cache: bool = False, **kwargs: Any
-    ) -> bool:
+        self, pair: bool, dangerous_use_bleak_cache: bool = False, **kwargs: Any
+    ) -> None:
         """
         Connect to a specified Peripheral.
 
-        **kwargs:
-            timeout (float): Timeout for required
-                ``BleakScanner.find_device_by_address`` call. Defaults to 10.0.
+        Args:
+            pair: If True, attempt to pair with the device after connecting.
+                  Note: Explicit pairing during connect is not available in ESPHome.
+                  Use the pair() method after connecting if pairing is needed.
+            dangerous_use_bleak_cache: Use cached services if available.
+            **kwargs:
+                timeout (float): Timeout for required
+                    ``BleakScanner.find_device_by_address`` call. Defaults to 10.0.
 
-        Returns
+        Returns:
         -------
             Boolean representing connection status.
 
         """
+        if pair:
+            _LOGGER.warning(
```

Comment:
A simple way to implement this could be to just `await self.pair()` between connecting and getting services.

Reply:
https://github.com/Bluetooth-Devices/bleak-esphome/pull/173